torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -3298,12 +3324,32 @@ def _call_function(
 class CheckpointHigherOrderVariable(WrapHigherOrderVariable):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        # If side effects are allowed under checkpoint, we should not restore
-        # the side effects after speculate subgraph.
+        # For checkpoint, restore_side_effects is the inverse of the config flag.
+        # When skip_fwd_side_effects_in_bwd is True, we don't restore (we allow side effects).
         self.restore_side_effects = (
             not torch._dynamo.config.skip_fwd_side_effects_in_bwd_under_checkpoint
         )
 
+    def create_wrapped_node(
+        self,
+        tx: "InstructionTranslator",
+        fn_vt,
+        fn_args_vt,
+        kwargs,
+        description,
+        *,
+        subgraph_name="wrap_body",
+    ):
+        # Set under_activation_checkpoint flag on tracer for checkpoint-specific logic
+        orig_val = tx.output.current_tracer.under_activation_checkpoint
+        try:
+            tx.output.current_tracer.under_activation_checkpoint = True
+            return super().create_wrapped_node(
+                tx, fn_vt, fn_args_vt, kwargs, description, subgraph_name=subgraph_name
+            )
+        finally:
+            tx.output.current_tracer.under_activation_checkpoint = orig_val
```

Comment:
I think we can get rid of `under_activation_checkpoint` entirely from the codebase. So need to overwrite this method.